azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/MavenAuthConfiguration.java

```diff
@@ -22,4 +28,39 @@
     private String serverId;
     private String httpProxyHost;
     private String httpProxyPort;
+
+    public void validate() throws InvalidConfigurationException {
+        String messagePrefix = null;
+        if (StringUtils.isBlank(tenant)) {
+            messagePrefix = "Cannot find 'tenant'";
+        } else if (StringUtils.isBlank(client)) {
+            messagePrefix = "Cannot find 'client'";
+        } else if (StringUtils.isAllBlank(key, certificate)) {
+            messagePrefix = "Cannot find either 'key' or 'certificate'";
+        } else if (StringUtils.isNotBlank(key) && StringUtils.isNotBlank(certificate)) {
+            messagePrefix = "It is wrong to specify both 'key' and 'certificate'";
+        } else if (StringUtils.isNotBlank(environment) && !AuthHelper.validateEnvironment(environment)) {
+            messagePrefix = "Invalid environment string '%s'";
+        } else if ((StringUtils.isNotBlank(httpProxyHost) && StringUtils.isBlank(httpProxyPort)) ||
+                (StringUtils.isBlank(httpProxyHost) && StringUtils.isNotBlank(httpProxyPort))) {
+            messagePrefix = "if you want to use proxy, 'httpProxyHost' and 'httpProxyPort' must both be set";
+        } else if (StringUtils.isNotBlank(httpProxyPort)) {
+            if (!StringUtils.isNumeric(httpProxyPort)) {
+                messagePrefix = String.format("Invalid integer number for httpProxyPort: '%s'", httpProxyPort);
+            } else if (NumberUtils.toInt(httpProxyPort) <= 0 || NumberUtils.toInt(httpProxyPort) > 65535) {
+                messagePrefix = String.format("Invalid range of httpProxyPort: '%s', it should be a number between %d and %d", httpProxyPort, 1, 65535);
+            }
+        }
+
+        final String messagePostfix = StringUtils.isNotBlank(serverId) ? (" in server: '" + serverId + "' at maven settings.xml.")
+                : " in <auth> configuration.";
+        if (Objects.nonNull(messagePrefix)) {
+            throw new InvalidConfigurationException(messagePrefix + messagePostfix);
+        }
+    }
```

Comment:
how about to use Preconditions.checkArgument(expression, message) here ?

Reply:
checkArgument will throw IllegalArgumentException only, we want to throw InvalidConfigurationException